libbeat/docs/shared-env-vars.asciidoc

```diff
@@ -49,3 +65,38 @@ and what each configuration looks like after replacement:
 |`name: ${NAME:beats}` |no setting            |`name: beats`
 |`name: ${NAME:beats}` |`export NAME=elastic` |`name: elastic`
 |==================================
```

Comment:
regarding this table, 2 syntax types are missing:

According [to this](https://github.com/elastic/go-ucfg/blob/master/variables.go#L68) we've got 3 kind of operators:

`:`  is default value operator
`:?` is the error message operator. If the environment variable can not be expanded, the error message will contain the custom error message... e.g. `name: ${NAME:?You need to set the NAME environment variable}` will have the error message contain the string `You need to set the NAME environment variable`, if `NAME` is not set.
`:+` is alternate value operator (if variable is set, expand with alternative value)
        e.g. name: ${NAME:+beats} with export NAME=elastic becomes `name: beats`. By nesting variable expansions with default cases and alternative rules one can build some kind of negation and checks. This is quite some advanced use-case almost no one will use in practice (it's somewhat tricky to use this one right). e.g. use case `include other string if variable is set`:

```
cmd: a b ${param:+-x ${param}}
```

will expand to `cmd: a b -x y` if `export param=y` and `cmd: a b` only if `param` is not set.